org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AbstractMatcher.java

```diff
@@ -129,6 +130,39 @@ final void nextIsSwitch() {
 		}
 	}
 
+	/**
+	 * Moves {@link #cursor} to next instruction if it is <code>LDC</code> with
+	 * the given <code>cst</code>, otherwise sets it to <code>null</code>.
+	 */
+	final void nextIsLdc(final String cst) {
+		nextIs(Opcodes.LDC);
+		if (cursor == null) {
+			return;
+		}
+		if (((LdcInsnNode) cursor).cst.equals(cst)) {
+			return;
+		}
+		cursor = null;
+	}
+
+	/**
+	 * Moves {@link #cursor} to next instruction if it is <code>LDC</code> with
+	 * <code>cst</code> that starts with <code>cstPrefix</code>, otherwise sets
+	 * it to <code>null</code>.
+	 */
+	final void nextIsLdcStartingWith(final String cstPrefix) {
+		nextIs(Opcodes.LDC);
+		if (cursor == null) {
+			return;
+		}
+		LdcInsnNode ldcNode = (LdcInsnNode) cursor;
+		if (ldcNode.cst instanceof String
+				&& ((String) ldcNode.cst).startsWith(cstPrefix)) {
```

Comment:
After `instanceOf` it does not recognise this type as `String` so you need an additional cast?

Reply:
yes :( the Java compiler doesn't have smart casts afaik...